### PR TITLE
Change SimpleNLPModel to use automatic differentiation

### PR DESCRIPTION
**Pull request by abelsiqueira**
This makes it much easier to create a simple model.
Since SimpleNLPModel is not intented for speed, this is good for the user.

### CODE REVIEW

**Comment by abelsiqueira** [comment]
Closes #15

File: src/simple_model.jl [hunk 1]
```diff
@@ -72,202 +43,79 @@ function SimpleNLPModel(x0::Vector, obj::Function; y0::Vector = [],
...
-    try
-      return nlp.Hprod!(x, w, v, Hv)
-    catch
-      Hv[:] = nlp.H(x,w) * v
-      return Hv
-    end
+    obj_weight = 1.0, y :: Vector = [])
+  Hv[:] = ForwardDiff.hessian(nlp.f, x) * v * obj_weight
+  for i = 1:length(y)
+    Hv[:] += ForwardDiff.hessian(x->nlp.c(x)[i], x) * v * y[i]
```
Comment (dpo) [inline src/simple_model.jl]: I think we could test whether `obj_weight` or any of the `y[i]` is zero here.

**Comment by abelsiqueira** [comment]
Updated

**Comment by dpo** [comment]
Looks great, thank you! Would you like to add the eval counters at this point?

**Comment by abelsiqueira** [comment]
I will add them and the tests in another PR, after the CUTEst PR, otherwise it will be too large

**Pull request merged**